test/distributed/algorithms/quantization/test_quantization.py

```diff
@@ -0,0 +1,158 @@
+import torch
+import os
+import torch.cuda
+import torch.distributed as dist
+import torch.distributed.algorithms.quantization.quantization as quant
+from torch.distributed.algorithms.quantization.quantization import DQuantType
+from torch.testing._internal.common_distributed import (
+    MultiProcessTestCase,
+    requires_gloo,
+    skip_if_lt_x_gpu,
+    requires_nccl,
+)
+from torch.testing._internal.distributed.distributed_test import (
+    apply_hack_for_nccl, BACKEND
+)
+from torch.testing._internal.common_utils import sandcastle_skip_if
+
+def _build_tensor(size, value=None, dtype=torch.float, device_id=None):
+    if value is None:
+        value = size
+    if device_id is None:
+        return torch.empty(size, size, size, dtype=dtype).fill_(value)
+    else:
+        return torch.empty(size, size, size, dtype=dtype).fill_(value).cuda(device_id)
+
+class DistQuantizationTests(MultiProcessTestCase):
+    def setUp(self):
+        super(DistQuantizationTests, self).setUp()
+        self._spawn_processes()
+
+    def tearDown(self):
+        super(DistQuantizationTests, self).tearDown()
+        try:
+            os.remove(self.file_name)
+        except OSError:
+            pass
+
+    @property
+    def op_timeout_sec(self):
+        return 1
+
+    @property
+    def world_size(self):
+        return 2
+
+    def _init_multigpu_helper(self):
+        """Multigpu tests are designed to simulate the multi nodes with multi
+        GPUs on each node. Nccl backend requires equal #GPUs in each process.
+        On a single node, all visible GPUs are evenly
+        divided to subsets, each process only uses a subset.
+        """
+        nGPUs = torch.cuda.device_count()
+        world_size = dist.get_world_size()
+        visible_devices = range(nGPUs)
+
+        if BACKEND == "nccl":
+            apply_hack_for_nccl()
+
+        # If rank is lesser than or equal to number of available GPU's
+        # then each rank can be mapped to corresponding GPU.
+        nGPUs_per_process = 1
+        if world_size > nGPUs:
+            nGPUs_per_process = nGPUs // world_size
+        rank_to_GPU = {
+            i: list(
+                visible_devices[i * nGPUs_per_process : (i + 1) * nGPUs_per_process]
+            )
+            for i in range(world_size)
+        }
+        return rank_to_GPU
+
+    @requires_gloo()
+    @sandcastle_skip_if(BACKEND != "gloo", "Only gloo backend supports all_gather_fp16")
+    def test_all_gather_fp16(self):
+        store = dist.FileStore(self.file_name, self.world_size)
+        dist.init_process_group(store=store, rank=self.rank, world_size=self.world_size, backend='gloo')
+        device = torch.device(f"cuda:{self.rank}")
+        group_id = dist.new_group(range(self.world_size))
```

Comment:
why we need to create a new_group here? we should use the global group instead, just like `_init_global_test` in `distributed_test.py` did?